tests/dist/python/run_dist_objects.py

```diff
@@ -48,6 +49,233 @@ def run_server(
     g.start()
 
 
+##########################################
+############### DistGraph ###############
+##########################################
+
+def dist_graph_test(g):
+    gpb_path = graph_path + '/{}.json'.format(graph_name)
+    with open(gpb_path) as conf_f:
+        part_metadata = json.load(conf_f)
+    assert 'num_nodes' in part_metadata
+    assert 'num_edges' in part_metadata
+    num_nodes = part_metadata['num_nodes']
+    num_edges = part_metadata['num_edges']
+
+    assert g.number_of_nodes() == num_nodes
+    assert g.number_of_edges() == num_edges
+
+    assert g.node_attr_schemes()['feat'].dtype == F.int64
+    assert g.node_attr_schemes()['feat'].shape == (1,)
+
+    # Test node_split
+
+    gpb = g.get_partition_book()
+
+    selected_nodes = np.random.randint(0, 100, size=g.number_of_nodes()) > 98
+
+    nodes = node_split(selected_nodes, gpb, force_even=False)
+    local_nids = gpb.partid2nids(gpb.partid, ntype=g.ntypes[0])
+
+    for n in nodes:
+        assert n in local_nids
+
+    nodes = node_split(selected_nodes, gpb, force_even=True)
+    subset = np.intersect1d(F.asnumpy(nodes), F.asnumpy(local_nids))
+    print('intersection has', len(subset))
+    selected_nids = np.nonzero(selected_nodes)[0]
+    for n in subset:
+        assert n in selected_nids
+
+    # Test edge_split
+
+    selected_edges = np.random.randint(0, 100, size=g.number_of_edges()) > 98
+    edges = edge_split(selected_edges, gpb, force_even=False)
+
+    local_eids = gpb.partid2eids(gpb.partid, etype=g.etypes[0])
+    for e in edges:
+        assert e in local_eids
+
+    edges = edge_split(selected_edges, gpb, force_even=True)
+    subset = np.intersect1d(F.asnumpy(edges), F.asnumpy(local_eids))
+    print('intersection has', len(subset))
+    selected_eids = np.nonzero(selected_edges)[0]
+    for e in subset:
+        assert e in selected_eids
+
+
+
+def dist_graph_hetero_test(g):
+    gpb = g.get_partition_book()
+
+    num_nodes = {'n1': 10000, 'n2': 10010, 'n3': 10020}
```

Comment:
we'd better avoid magic numbers. can we obtain such info from original graph or passed in arguments?